includes/validate.php

```diff
@@ -184,65 +184,47 @@ function edac_validate( $post_ID, $post, $action ) {
 function edac_remove_corrected_posts( $post_ID, $type, $pre = 1, $ruleset = 'php' ) {
 	global $wpdb;
 
-	$rules        = edac_register_rules();
-	$js_rule_ids  = array();
-	$php_rule_ids = array();
+	$rules      = edac_register_rules();
+	$rule_slugs = array();
 	foreach ( $rules as $rule ) {
-		if ( array_key_exists( 'ruleset', $rule ) && 'js' === $rule['ruleset'] ) {
-			$js_rule_ids[] = $rule['slug'];
-		} else {
-			$php_rule_ids[] = $rule['slug'];
+		if ( 'js' === $ruleset && isset( $rule['ruleset'] ) && 'js' === $rule['ruleset'] ) {
+			$rule_slugs[] = $rule['slug'];
+		} elseif ( 'js' !== $ruleset ) {
+			$rule_slugs[] = $rule['slug'];
 		}
```

Comment:
Instead of compiling 2 different arrays, one for JS and one for PHP, this changes the code to only compile a single array with the rules we need. We already know if we need PHP or JS rules since it's passed to the function as `$ruleset`, so it's a lot cleaner to have a single array and then use that in the query. This also helps simplify the code below for queries.